gk/main.c

```diff
@@ -1986,7 +2014,13 @@ gk_proc(void *arg)
 
 		now = rte_rdtsc();
 		if (now - last_scan_tsc >= bucket_scan_timeout_cycles) {
-			gk_flow_tbl_bucket_scan(&bucket_idx, instance);
+			/*
+			 * Reset the flag @has_insertion_failed when
+			 * one or more entries have been freed from
+			 * the flow table.
+			 */
+			instance->has_insertion_failed =
+				!gk_flow_tbl_bucket_scan(&bucket_idx, instance);
```

Comment:
Rewrite the line above as follows:
```
if (gk_flow_tbl_bucket_scan(&bucket_idx, instance))
	instance->has_insertion_failed = false;
```
Otherwise `has_insertion_failed` can become `true` just because there's no expired entry in the bucket.